### PR TITLE
gpio: gpio_keys_zephyr: add missing gpio initialization

### PR DESCRIPTION
**Pull request by fabiobaltieri**
The driver is missing the GPIO initialization entirely, meaning that flags like PULL_UP are not currently being applied. Add the missing call.

### CODE REVIEW

File: drivers/gpio/gpio_keys_zephyr.c [hunk 1]
```diff
@@ -192,8 +192,16 @@ static int gpio_keys_zephyr_get_pin(const struct device *dev, uint32_t idx)
 static int gpio_keys_init(const struct device *dev)
 {
 	struct gpio_keys_data *data = dev->data;
+	const struct gpio_keys_config *cfg = dev->config;
+	int ret;
 
 	for (int i = 0; i < data->num_keys; i++) {
+		ret = gpio_pin_configure_dt(&cfg->pin_cfg[i].spec, GPIO_INPUT);
```
Comment (bbilas) [inline drivers/gpio/gpio_keys_zephyr.c:199]: You should call `gpio_is_ready_dt` before this.

Comment (fabiobaltieri) [reply drivers/gpio/gpio_keys_zephyr.c:199]: Good point thanks, did not realize we had a specialized gpio+dt one for that, good stuff.

**Review by asemjonovs — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Comment by cfriedt** [comment]
Technically fixes a bug - I'll leave it to you if you want to file an issue.

**Comment by fabiobaltieri** [comment]
> Technically fixes a bug - I'll leave it to you if you want to file an issue.

I don't think we want to start flooding the issue queue with bugs that don't need triaging, maybe let's pick the conversation back up tomorrow.

**Review by stephanosio — APPROVED** [review]

**Review by bbilas — APPROVED** [review]

**Pull request merged**